src/docs/sdk/performance/ui-event-transactions.mdx

```diff
@@ -0,0 +1,160 @@
+    And the transaction has two running child spans
+    When the SDK finishes the first child span
+    Then the SDK doesn't reset the idle timeout
+    And waits for the children to finish
+
+Scenario: Discard UI event transactions without child spans
+    Given a UI event transaction
+    And the transaction has no child spans
+    When the idleTimeout times out
+    Then the SDK discards the transaction 
+
+Scenario: Set time to last finished child span
+    Given a UI event transaction
+    And the transaction has one finished child span
+    And the transaction has one running child span
+    When the running child span finishes
+    Then the SDK finishes the transaction
+    And trims the end time of the transaction to the one of the last finished
+        child span
+
+Scenario: Same UI element with same event
+    Given an ongoing UI event transaction
+    When the user triggers the same UI event with the same type for the same
+        UI element
+    Then the SDK resets the timeout
+    And doesn't create a new transaction
+
+# If your SDK binds auto-genrated transactions to the scope see binding
+# to scope
+Scenario: Same UI element with different event
+    Given an ongoing UI event transaction
+    When the user triggers the same UI element with a different event
+    Or the user triggers a different UI element
+    Then the SDK finishes the ongoing transaction
+    And sets the status to OK
+    And waits for the children to finish
+    And cancels the timeout
+    And starts a new transaction
+```
+
+## Binding to scope
+
+On platforms mainly interacting with the static API, such as mobile, it's typical to bind auto-generated
+transactions to the scope so that users can access them via the static API. We recommend binding UI event
+transactions to the scope on these platforms. The following extra rules apply as UI event transactions
+could interfere with other auto-generated transactions.
+
+```Gherkin
+Scenario: Same UI element with different event
+    Given an ongoing UI event transaction
+    When the user triggers the same UI element with a different event
+    Or the user triggers a different UI element
+    Then the SDK finishes the ongoing transaction
+    And sets the status to OK
+    And waits for the children to finish
+    And cancels the timeout
+    And removes the ongoing transaction from the scope
+    And starts a new transaction
+    And puts the new transaction on the scope
+
+Scenario: UI event triggered but transaction ended
+    Given an auto-generated transaction from a any UI event
+    And the transaction already  is finished
+    When the user triggers the same UI event
+    Then the SDK starts a new UI event transaction
+
+Scenario: Manually created transaction bound to the scope
+    Given an ongoing manually created transaction by the user bound to the
+        scope
+    When the SDK creates a new UI event transaction
+    Then the SDK doesn't add the UI event transaction to the scope 
```

Comment:
In this case, if the transaction is not in the scope, there will be no spans for it since the user can't access it nor the auto instrumentation, doesn't that mean that the transaction won't have spans and won't be sent?

Reply:
Basically, yes, we could drop it and not start it. That's an improvement. Good catch, thanks 🙏 . @romtsn, I think we could change that in our SDKs. WDYT?

Reply:
I guess it makes sense if it's not an Idle transaction (for platforms that you know that the loading is finished), but not with the idle transaction concept.

Reply:
yeah we could do that, I guess that would work for any kind of transaction on the scope - if there's something, we just don't start the UI transaction at all

Reply:
Should we raise a meta issue for that?

Reply:
https://github.com/getsentry/team-mobile/issues/71